youtube_dl/extractor/itunes.py

```diff
@@ -0,0 +1,59 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+
+from ..utils import (
+    extract_attributes,
+    int_or_none,
+    unescapeHTML,
+    unified_strdate,
+)
+
+
+class iTunesIE(InfoExtractor):
+    _VALID_URL = r'https?://itunes\.apple\.com/[a-z]{2}/[a-z0-9-]+/(?P<display_id>[a-z0-9-]+)?/(?:id)?(?P<id>[0-9]+)'
```

Comment:
It should be the following as the (valid) URL variations of https://itunes.apple.com/us/itunes-u/uc-davis-symphony-orchestra/id403834767 don't match otherwise

``` python
_VALID_URL = r'https?://itunes\.apple\.com/[a-z]{2}?/?[a-z0-9-]+/?(?P<display_id>[a-z0-9-]+)?/(?:id)?(?P<id>[0-9]+)'
```

* https://itunes.apple.com/itunes-u/id403834767
* https://itunes.apple.com/us/itunes-u/id403834767
* https://itunes.apple.com/itunes-u/uc-davis-symphony-orchestra/id403834767